src/instructionFiles/InstructionFileManager.ts

```diff
@@ -0,0 +1,311 @@
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as path from 'path';
+import {
+  NPL_INSTRUCTION_VERSION,
+  NPL_SECTION_START_MARKER,
+  NPL_SECTION_END_MARKER,
+  COPILOT_INSTRUCTIONS_PATH,
+  CURSOR_RULES_PATH,
+  RESOURCES_DIR,
+  TEMPLATES_DIR,
+  NPL_INSTRUCTIONS_TEMPLATE_FILENAME
+} from '../constants';
+
+export enum DialogButton {
+  Yes = 'Yes',
+  Always = 'Always & auto-update',
+  AlwaysUpdate = 'Always',
+  Never = 'Never'
+}
```

Comment:
these control a single setting. We *could* have two different settings for creation and updating, but I thought it was overkill for now...